httpsender/greenbone-maintain-auth.js

```diff
@@ -0,0 +1,131 @@
+// Greenbone auth & sessions are a bit flaky with scans ...
+// ... this helps make auth less flaky
+if (typeof println == 'undefined') this.println = print;
+
+// Logging with the script name is super helpful!
+function logger() {
+  print('[' + this['zap.script.name'] + '] ' + arguments[0]);
+}
+
+function dropMessage(msg) {
+  msg.setResponseBody("<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2//EN\">\n" +
+      "<html><head></head><body><h1>403 Forbidden</h1>\n" +
+      "Out of scope request blocked by ZAP script 'Drop requests not in scope.js'\n" +
+      "</body></html>");
+  msg.setResponseHeader("HTTP/1.1 403 Forbidden\r\n" +
+      "Content-Type: text/html; charset=UTF-8");
+  msg.getResponseHeader().setContentLength(msg.getResponseBody().length());
+}
+
+function isStaticUrl(url) {
+  if (url.indexOf('.xml') !== -1) {
+    return true;
+  }
+
+  if (url.indexOf('.css') !== -1) {
+    return true;
+  }
+  
+  if (url.indexOf('.gif') !== -1) {
+    return true;
+  }
+
+  if (url.indexOf('.js') !== -1) {
+    return true;
+  }
+  
+  if (url.indexOf('.txt') !== -1) {
+    return true;
+  }
+
+  if (url.indexOf('.htm') !== -1) {
+    return true;
+  }
+  return false;
+}
+
+var COOKIE_TYPE   = org.parosproxy.paros.network.HtmlParameter.Type.cookie;
+var ScriptVars    = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
+var HtmlParameter = Java.type('org.parosproxy.paros.network.HtmlParameter');
+
+// Rewrite requests to include correct query token param
+function sendingRequest(msg, initiator, helper) {  
+  var reqbody = msg.getRequestBody().toString();
+  var headers = msg.getRequestHeader();
+  var url     = headers.getURI().toString();
+  var qry     = headers.getURI().getQuery();
+  var cookies = headers.getCookieParams();
+  
+  if (initiator === HttpSender.SPIDER_INITIATOR) {}
+  if (isStaticUrl(url)) {return true;}
```

Comment:
The return values should be removed, the HTTP Sender scripts don't return anything.